src/staticdata.c

```diff
@@ -1205,7 +1205,7 @@ static void jl_save_system_image_to_stream(ios_t *f)
         }
     }
 
-    jl_idtable_type = jl_base_module ? jl_get_global(jl_base_module, jl_symbol("ObjectIdDict")) : NULL;
+    jl_idtable_type = jl_base_module ? jl_get_global(jl_base_module, jl_symbol("_IdDict")) : NULL;
```

Comment:
This then uses https://github.com/JuliaLang/julia/pull/25210/files#r158050304

Reply:
The rehashing will now need to be done for all instantiations of `IdDict`, not just `IdDict{Any,Any}`.

Reply:
But just replacing it with `jl_symbol("IdDict")` does not work.  Julia compiles fine but the parallel stuff does not work (and thus the test don't run either). On eac2ff7:
```
julia> using Distributed; addprocs(2)
      From worker 2:    Master process (id 1) could not connect within 60.0 seconds.Worker 2 terminated.
ERROR: 
      From worker 2:    exiting.StackOverflowError:
haskey at ./abstractdict.jl:553 [inlined]
is_root_module at ./loading.jl:371 [inlined]
serialize_mod_names(::Distributed.ClusterSerializer{TCPSocket}, ::Module) at ./serialize.jl:355
serialize_mod_names(::Distributed.ClusterSerializer{TCPSocket}, ::Module) at ./serialize.jl:358 (repeats 97 times)

...and 1 more exception(s).


Stacktrace:ERROR (unhandled task failure): Version read failed. Connection closed by peer.

 [1] sync_end() at ./task.jl:300
 [2] macro expansion at ./task.jl:316 [inlined]
 [3] #addprocs_locked#40(::NamedTuple{(),Tuple{}}, ::Function, ::Distributed.LocalManager) at /home/mauro/julia/julia-master/usr/share/julia/site/v0.7/Distributed/src/cluster.jl:401
 [4] addprocs_locked at /home/mauro/julia/julia-master/usr/share/julia/site/v0.7/Distributed/src/cluster.jl:372 [inlined]
 [5] #addprocs#39(::NamedTuple{(),Tuple{}}, ::Function, ::Distributed.LocalManager) at /home/mauro/julia/julia-master/usr/share/julia/site/v0.7/Distributed/src/cluster.jl:365
 [6] addprocs at /home/mauro/julia/julia-master/usr/share/julia/site/v0.7/Distributed/src/cluster.jl:361 [inlined]
 [7] #addprocs#268 at /home/mauro/julia/julia-master/usr/share/julia/site/v0.7/Distributed/src/managers.jl:316 [inlined]
 [8] addprocs(::Int64) at /home/mauro/julia/julia-master/usr/share/julia/site/v0.7/Distributed/src/managers.jl:315
 [9] top-level scope
```

I think the problem is the re-hashing:
```
julia> haskey(Base.module_keys, Main) # true about half the time
false

julia> Base.rehash!(Base.module_keys);

julia> haskey(Base.module_keys, Main)
true
```